ivadomed/testing.py

```diff
@@ -170,14 +170,14 @@ def run_inference(test_loader, model, model_params, testing_params, ofolder, cud
                 preds_idx_arr = np.array(preds_idx_undo)
 
                 # TODO: gt_filenames should not be a list
-                fname_ref = metadata_idx[0]['gt_filenames'][0]
+                fname_ref = list(filter(None, metadata_idx[0]['gt_filenames']))[0]
 
                 # NEW COMPLETE VOLUME
                 if pred_tmp_lst and (fname_ref != fname_tmp or last_sample_bool) and task != "classification":
                     # save the completely processed file as a nifti file
                     if ofolder:
                         fname_pred = os.path.join(ofolder, fname_tmp.split('/')[-1])
-                        fname_pred = fname_pred.rsplit(testing_params['target_suffix'][0],1)[0] + '_pred.nii.gz'
+                        fname_pred = fname_pred.rsplit("_", 1)[0] + '_pred.nii.gz'
```

Comment:
In which case `target_suffix` was incorrect?
I am not sure if `.rsplit("_", 1)[0]` will always be robust :/

Reply:
we take the first target_suffix (`target_suffix[0]), but it might be the second target suffix in fname_pred, if the first target doesn't exist. If the folder respects BIDS conventions, this line will work (remove target_suffix et replace by _pred.nii.gz). I think we can assume the files respect this convention.

Reply:
I was confused with `rsplit` behaviour, made some tests: you are right, all good :-)